pkg/controllers/allocation/suite_test.go

```diff
@@ -138,15 +140,18 @@ var _ = Describe("Allocation", func() {
 
 			nodes := &v1.NodeList{}
 			Expect(env.Client.List(ctx, nodes)).To(Succeed())
-			Expect(len(nodes.Items)).To(Equal(6)) // 5 schedulable -> 5 node, 2 coschedulable -> 1 node
 			for _, pod := range schedulable {
 				scheduled := ExpectPodExists(env.Client, pod.GetName(), pod.GetNamespace())
 				ExpectNodeExists(env.Client, scheduled.Spec.NodeName)
 			}
 			for _, pod := range unschedulable {
 				unscheduled := ExpectPodExists(env.Client, pod.GetName(), pod.GetNamespace())
+				if unscheduled.Spec.NodeName != "" {
+					fmt.Printf("This pod shouldn't have been scheduled: %s", pretty.Concise(pod))
+				}
```

Comment:
This might just be stray debugging code left in?

Reply:
Do you have a problem with it? It's a test, this helps people figure out which test failed. I have no idea how someone is supposed to debug this test failure without this.

Reply:
This comment also feels inflammatory as I re-read it. What I really wanted to say was: is there some conventional reason that it's wrong to print additional debugging information in this way for a test failure? Is there a preferred convention in Go? Without this change, I honestly don't know how to debug this test failure.

Reply:
The tests are designed around Expect() calls. If any of the expectations fail, then the test will fail loudly with that message.

Expect(unscheduled.Spec.NodeName).ToNot(BeEmpty())
// logline is: "expected unscheduled.Spec.NodeName to not be empty, but it was _____"